packages/foam-vscode/src/services/templates.ts

```diff
@@ -105,6 +113,86 @@ export type OnFileExistStrategy =
   | 'ask'
   | ((filePath: URI) => Promise<URI | undefined>);
 
+async function askUserForTemplate() {
```

Comment:
I think it makes sense to move this function here, good call.
I would remove the old one and have this one being the only one in use

Reply:
(same comment applies to the other functions here)